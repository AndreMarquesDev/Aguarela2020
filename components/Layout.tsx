/* eslint-disable no-console */
import React, { FC } from 'react';
import { useRouter } from 'next/router';
import GeneralStyles from '../styles/styles';
import Header from './Header';
import colors from '../styles/colors';
import LanguageButton from './LanguageButton';
import { bodyPadding } from '../styles/variables';

const Layout: FC = props => {
    const { children } = props;

    const {
        route,
        query,
    } = useRouter();

    return (
        <main className="Layout">
            <GeneralStyles />

            <Header currentRoute={route} language={query.language?.toString()} />

            {children}

            <LanguageButton language={query.language?.toString()} />

            <style jsx>
                {`
                    .Layout {
                        position: relative;
                        padding: ${bodyPadding} 0;

                        &:before {
                            content: '';
                            background: ${colors.lightGrey};
                            width: 50%;
                            height: 100%;
                            position: absolute;
                            top: 0;
                            left: 0;
                            z-index: -1;
                        }
                    }
                `}
            </style>

            {/* {console.log('%c| üîß Developed by AndreMarquesDev ‚úèÔ∏è Designed by Aguarela Digital |', 'background: #000; color: #fff;')}
            {console.log('%c| https://github.com/AndreMarquesDev |', 'background: #000; color: #fff;')}
            {console.log('%c| https://www.instagram.com/aguareladigital |', 'background: #000; color: #fff;')} */}
        </main>
    );
};

export default Layout;
