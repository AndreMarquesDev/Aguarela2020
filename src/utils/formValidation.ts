export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
