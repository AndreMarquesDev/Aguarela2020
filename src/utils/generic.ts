import type { TextsInterface } from './texts';
import { Locale } from '../types/Locale';
import { textsEn, textsPt } from './texts';

export const isDev = process.env.NODE_ENV === 'development';
export const capitalize = (string: string): string =>
    string.charAt(0).toUpperCase() + string.slice(1);
export const getCurrentLanguagetexts = (language: Locale): TextsInterface =>
    language === Locale.Pt ? textsPt : textsEn;

export const resetTimesTouchedAttribute = (): void =>
    document.querySelectorAll('[data-times-touched]').forEach(
        element => ((element as HTMLLIElement).dataset.timesTouched = '0'),
    );
