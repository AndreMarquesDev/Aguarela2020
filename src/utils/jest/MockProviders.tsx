/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
