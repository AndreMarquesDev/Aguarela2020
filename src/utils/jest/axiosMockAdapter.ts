import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { contactFormBackendUrl } from '../urls';

export const initializeAxiosMockAdapter = (delay = 0, isSuccess = true): void => {
    const mock = new MockAdapter(axios, {
        delayResponse: delay,
    });

    const httpStatusCode = isSuccess ? 200 : 500;

    mock.onPost(contactFormBackendUrl).reply(httpStatusCode, {});
};
