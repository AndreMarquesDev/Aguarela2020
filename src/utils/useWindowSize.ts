import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
