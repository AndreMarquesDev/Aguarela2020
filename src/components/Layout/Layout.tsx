import type { ReactNode } from 'react';
import type { TextsContextProps } from '../context/TextsContext';
import classNames from 'classnames';
import { useRouter } from 'next/router';
import React, { useContext, useMemo } from 'react';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../../utils/texts';
import { Breakpoint, useWindowSize } from '../../utils/useWindowSize';
import { NavLinksContext } from '../context/NavLinksContext';
import { TextsContext } from '../context/TextsContext';
import { Footer } from '../Footer/Footer';
import { Header } from '../Header/Header';

export type LayoutProps = {
    children: ReactNode;
};

export const Layout = ({ children }: LayoutProps): JSX.Element => {
    const { route, query } = useRouter();
    const { isMenuOpen } = useContext(NavLinksContext);

    const currentLanguage = query.language?.toString() as Locale;
    const texts = currentLanguage === Locale.Pt ? textsPt : textsEn;

    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Tablet;

    const textsContextValue = useMemo<TextsContextProps>(
        () => ({
            texts,
        }),
        [texts],
    );

    return (
        <>
            <TextsContext.Provider value={textsContextValue}>
                <main className={classNames('layout', isMenuOpen && !isDesktop && 'menuOpen')}>
                    <Header currentRoute={route} language={currentLanguage} />

                    {children}

                    <Footer language={currentLanguage} />
                </main>
            </TextsContext.Provider>

            <style jsx>
                {`
                    .layout {
                        position: relative;
                    }

                    .menuOpen {
                        position: fixed;
                        overflow: hidden;
                        width: 100%;
                    }
                `}
            </style>
        </>
    );
};
