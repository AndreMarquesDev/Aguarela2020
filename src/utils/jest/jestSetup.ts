// eslint-disable-next-line import/no-extraneous-dependencies
import mockRouter from 'next-router-mock';
import { Breakpoint } from '../useWindowSize';
import { setJestWindowWidth } from './setJestWindowWidth';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
    })),
});

setJestWindowWidth(Breakpoint.Desktop);

// eslint-disable-next-line import/no-extraneous-dependencies, ts/no-require-imports
jest.mock('next/router', () => require('next-router-mock'));

mockRouter.query = {
    language: 'pt',
};
