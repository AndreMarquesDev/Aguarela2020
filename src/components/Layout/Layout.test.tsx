import type { RenderResult } from '@testing-library/react';
import type { LayoutProps } from './Layout';
import { render, screen } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import React from 'react';
import { MockProviders } from '../../utils/jest/MockProviders';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { textsEn } from '../../utils/texts';
import { Breakpoint } from '../../utils/useWindowSize';
import { Layout } from './Layout';
import '@testing-library/jest-dom';

const baseProps: LayoutProps = {
    children: (
        <div>
            <p>Child</p>
        </div>
    ),
};

const renderComponent = (isMenuOpen = false): RenderResult => {
    return render(
        <MockProviders isMenuOpen={isMenuOpen}>
            <Layout {...baseProps} />
        </MockProviders>,
    );
};

describe('<Layout />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        const childElement = screen.getByText('Child');
        const logo = screen.getByAltText('Logo');

        expect(childElement).toBeInTheDocument();
        expect(logo).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly when the menu is open and the resolution is lower than desktop', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent(true);

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        mockRouter.query = {
            language: 'en',
        };

        const { container } = renderComponent();

        const aboutLink = screen.getByText(textsEn.about);

        expect(aboutLink).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
