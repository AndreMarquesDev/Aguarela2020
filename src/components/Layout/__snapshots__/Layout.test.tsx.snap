// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders properly 1`] = `
<div>
  <main
    class="jsx-1819645925 layout"
  >
    <header
      class="jsx-2405765474"
      data-testid="header_section"
    >
      <nav
        class="jsx-2405765474 wrapper"
      >
        <a
          data-testid="header_homepageLogoLink"
          href="/pt/homepage"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="35"
            loading="lazy"
            src="/images/logo.svg"
            style="color: transparent;"
            width="250"
          />
        </a>
        <ul
          class="jsx-2513558763 "
        >
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/about"
            >
              Sobre
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/services"
            >
              Serviços
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/contact"
            >
              Contacto
            </a>
          </li>
          <li
            class="jsx-2513558763 languageButton fontXXS"
          >
            <button
              class="jsx-2866958827"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2279021353 menuIcon hidden"
          data-testid="menuIcon_button"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2279021353"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-2748243618"
      data-testid="footer_section"
    >
      <div
        class="jsx-2748243618 wrapper"
      >
        <p
          class="jsx-2748243618 fontXS"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-2748243618 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.eu"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-2748243618"
        >
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-2748243618 iconLink"
              href="https://www.instagram.com/aguareladigital/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-2748243618 iconLink"
              href="https://www.facebook.com/aguareladigitalpt"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Go to contact page"
              class="iconLink"
              href="/pt/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;

exports[`<Layout /> renders properly in English 1`] = `
<div>
  <main
    class="jsx-1819645925 layout"
  >
    <header
      class="jsx-2405765474"
      data-testid="header_section"
    >
      <nav
        class="jsx-2405765474 wrapper"
      >
        <a
          data-testid="header_homepageLogoLink"
          href="/en/homepage"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="35"
            loading="lazy"
            src="/images/logo.svg"
            style="color: transparent;"
            width="150"
          />
        </a>
        <ul
          class="jsx-2513558763 mobileLayout"
        >
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/en/about"
            >
              About
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/en/projects"
            >
              Projects
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/en/services"
            >
              Services
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/en/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="jsx-2513558763 languageButton fontXXS white fontXS"
          >
            <button
              class="jsx-2866958827"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2279021353 menuIcon"
          data-testid="menuIcon_button"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2279021353"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-2748243618"
      data-testid="footer_section"
    >
      <div
        class="jsx-2748243618 wrapper"
      >
        <p
          class="jsx-2748243618 fontXS"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-2748243618 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.eu"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-2748243618"
        >
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-2748243618 iconLink"
              href="https://www.instagram.com/aguareladigital/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-2748243618 iconLink"
              href="https://www.facebook.com/aguareladigitalpt"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Go to contact page"
              class="iconLink"
              href="/en/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;

exports[`<Layout /> renders properly when the menu is open and the resolution is lower than desktop 1`] = `
<div>
  <main
    class="jsx-1819645925 layout menuOpen"
  >
    <header
      class="jsx-2405765474"
      data-testid="header_section"
    >
      <nav
        class="jsx-2405765474 wrapper"
      >
        <a
          data-testid="header_homepageLogoLink"
          href="/pt/homepage"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="35"
            loading="lazy"
            src="/images/logo.svg"
            style="color: transparent;"
            width="150"
          />
        </a>
        <ul
          class="jsx-2513558763 mobileLayout menuOpen"
        >
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/about"
            >
              Sobre
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/projects"
            >
              Projetos
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/services"
            >
              Serviços
            </a>
          </li>
          <li
            class="jsx-2513558763"
          >
            <a
              class="fontS navLinksLink"
              href="/pt/contact"
            >
              Contacto
            </a>
          </li>
          <li
            class="jsx-2513558763 languageButton fontXXS white fontXS"
          >
            <button
              class="jsx-2866958827"
              type="button"
            >
              pt
            </button>
          </li>
        </ul>
        <i
          aria-label="toggle menu"
          class="jsx-2279021353 menuIcon open"
          data-testid="menuIcon_button"
          role="button"
          tabindex="0"
        >
          <span
            class="jsx-2279021353"
          />
        </i>
      </nav>
    </header>
    <div>
      <p>
        Child
      </p>
    </div>
    <footer
      class="jsx-2748243618"
      data-testid="footer_section"
    >
      <div
        class="jsx-2748243618 wrapper"
      >
        <p
          class="jsx-2748243618 fontXS"
        >
          Aguarela Digital - Developed by
           
          <a
            class="jsx-2748243618 animatedLink animatedLinkWhite"
            href="https://www.andremarquesdev.eu"
            rel="noreferrer"
            target="_blank"
          >
            André Marques
          </a>
        </p>
        <ul
          class="jsx-2748243618"
        >
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela instagram"
              class="jsx-2748243618 iconLink"
              href="https://www.instagram.com/aguareladigital/"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Aguarela facebook"
              class="jsx-2748243618 iconLink"
              href="https://www.facebook.com/aguareladigitalpt"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                />
              </svg>
            </a>
          </li>
          <li
            class="jsx-2748243618"
          >
            <a
              aria-label="Go to contact page"
              class="iconLink"
              href="/pt/contact"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
    </footer>
  </main>
</div>
`;
